app/src/main/java/com/adam/aslfms/util/MD5.java

```diff
@@ -1,742 +1,54 @@
-		}
-
-		/**
-		 * Set this state to be exactly the same as some other.
-		 *
-		 * @param from state to copy from.
-		 *
-		 * @since ostermillerutils 1.00.00
-		 */
-		private void copy(MD5State from) {
-			System.arraycopy(from.buffer, 0, this.buffer, 0, this.buffer.length);
-			System.arraycopy(from.state, 0, this.state, 0, this.state.length);
-			this.valid = from.valid;
-			this.bitCount = from.bitCount;
-		}
-	}
-
-
-	/**
-	 * Turns array of bytes into string representing each byte as
-	 * a two digit unsigned hex number.
-	 *
-	 * @param hash Array of bytes to convert to hex-string
-	 * @return  Generated hex string
-	 *
-	 * @since ostermillerutils 1.00.00
-	 */
-	private static String toHex(byte hash[]){
-		StringBuffer buf = new StringBuffer(hash.length * 2);
-		for (byte element: hash) {
-			int intVal = element & 0xff;
-			if (intVal < 0x10){
-				// append a zero before a one digit hex
-				// number to make it two digits.
-				buf.append("0");
+			StringBuilder hexString = new StringBuilder();
+			for (byte aMessageDigest : messageDigest) {
+				String h = Integer.toHexString(0xFF & aMessageDigest);
+				while (h.length() < 2) {
+					h = "0" + h;
+				}
+				hexString.append(h);
 			}
-			buf.append(Integer.toHexString(intVal));
-		}
-		return buf.toString();
-	}
+			output =  hexString.toString();
 
-	private static int FF (int a, int b, int c, int d, int x, int s, int ac) {
-		a += ((b & c) | (~b & d));
-		a += x;
-		a += ac;
-		//return rotateLeft(a, s) + b;
-		a = (a << s) | (a >>> (32 - s));
-		return a + b;
-	}
-
-	private static int GG (int a, int b, int c, int d, int x, int s, int ac) {
-		a += ((b & d) | (c & ~d));
-		a += x;
-		a += ac;
-		//return rotateLeft(a, s) + b;
-		a = (a << s) | (a >>> (32 - s));
-		return a + b;
-	}
-
-	private static int HH (int a, int b, int c, int d, int x, int s, int ac) {
-		a += (b ^ c ^ d);
-		a += x;
-		a += ac;
-		//return rotateLeft(a, s) + b;
-		a = (a << s) | (a >>> (32 - s));
-		return a + b;
-	}
-
-	private static int II (int a, int b, int c, int d, int x, int s, int ac) {
-		a += (c ^ (b | ~d));
-		a += x;
-		a += ac;
-		//return rotateLeft(a, s) + b;
-		a = (a << s) | (a >>> (32 - s));
-		return a + b;
-	}
-
-	private static byte[] encode(long l){
-		byte[] out = new byte[8];
-		out[0] = (byte) (l & 0xff);
-		out[1] = (byte) ((l >>> 8) & 0xff);
-		out[2] = (byte) ((l >>> 16) & 0xff);
-		out[3] = (byte) ((l >>> 24) & 0xff);
-		out[4] = (byte) ((l >>> 32) & 0xff);
-		out[5] = (byte) ((l >>> 40) & 0xff);
-		out[6] = (byte) ((l >>> 48) & 0xff);
-		out[7] = (byte) ((l >>> 56) & 0xff);
-		return out;
-	}
-
-	private static byte[] encode(int input[], int len){
-		byte[] out = new byte[len];
-		int i, j;
-		for (i = j = 0; j  < len; i++, j += 4) {
-			out[j] = (byte) (input[i] & 0xff);
-			out[j + 1] = (byte) ((input[i] >>> 8) & 0xff);
-			out[j + 2] = (byte) ((input[i] >>> 16) & 0xff);
-			out[j + 3] = (byte) ((input[i] >>> 24) & 0xff);
+		} catch (NoSuchAlgorithmException e) {

```

Comment:
I'd suggest just rethrowing this as a runtime exception. Unless there's something sensible we can do if MD5 is not available? Is it ever not available?